packages/grid/_modules_/grid/hooks/features/useGridColumnResize.tsx

```diff
@@ -179,7 +188,7 @@ export const useGridColumnResize = (
 
```

Comment:
you could add a `setColumnWidth` api to make it nicer. You should also publish an event like ColumnHeaderWidth change...

Reply:
I was thinking about that (it is listed in the ticket), but my argument against it is that you can already do that using the state. Is there a value to have a dedicated API just for this?

Reply:
Yes it's to make the api more friendly to use. The width is something that can get updated quite a lot. It's good to have an abstraction for beginners.

Reply:
Yes but it isn't difficult to achieve. Basically `setColumnWidth` will look like this:

```tsx
const setColumnWidth = (field: string, width: number) => {
  const column = apiRef.current.getColumnFromField(field);
  apiRef.current.updateColumn({ ...column, width});
}
```

Also if that is the case I think this would be better in the `useGridColumns` hook.

Reply:
Yes if we can consider it a core columns function then it could go to `useGridColumns`.
Let's do it :)

Reply:
So what's up with the api function?

Reply:
Added, the thing is that in the `useColumnResize` manipulating the dom directly is still used because the performance is much smoother.